### DISCUSSION TITLE
How do I get notified of state changes of a sleepy contact sensor programmatically

### DISCUSSION THREAD

**Discussion by plackettsj1** [openthread/openthread#9796] [Q&A] [answered]
Hi, I have setup up a network with 4 TBR's which are combination of RPi and Nordic Semiconductor NCP/RCP using OTBR instructions/software. I have joined these devices to an Amazon Alexa Gen 4 using a nanoleaf bulb to gain the network key. All of this works well, they have formed a single network. 

I have now attached an Eve Door Sensor (Matter Enabled) and can see it in the topology and the Alexa App can see its state (Open/Closed) I would like to get hold of the state using an API or using UDP, or any other programmatic method. I apologise in advance if I have missed something, but I can't seem to find out how to register for the state changes that the Eve Sensor reports. 

I have been unable to decipher the wire shark info and decided I must be looking in the wrong place, digging this deep!!
 
I think I can see UDP messages being sent from the logs on the home made TBRs, but can't seem to listen for them. I have tried listening to ports 5540 and 5541 on the IPv6 network but can't pick up anything. 

I can find the Eve sensors unique identifier which appears to be a unique v6 ip addr, as I can get on to the TBR it has attached to and get hold of its info from the child table etc.

Please could you point me to somewhere that I can find help ?

Apologies in advance if I have missed something obvious
Thanks
Simon

**Comment by jwhui** [comment, marked as answer]
The Eve device implements Matter, so you'll need to interact with it using the Matter protocol. Note that Matter implements application-layer, end-to-end security, so you'll also need to properly provision security credentials to effectively interact with the device. Learn more about Matter here: https://csa-iot.org/all-solutions/matter/
